assets/samples/dynakube/v1beta4/kubernetesSecurityPostureManagement.yaml

```diff
@@ -3,31 +3,195 @@ kind: DynaKube
 metadata:
   name: dynakube
   namespace: dynatrace
+    # annotations:
+  # feature.dynatrace.com/k8s-app-enabled: "true"
 spec:
+  # Link to api reference for further information: https://docs.dynatrace.com/docs/ingest-from/setup-on-k8s/reference/dynakube-parameters
+
+  # Required: Dynatrace apiUrl including the `/api` path at the end.
   apiUrl: https://ENVIRONMENTID.live.dynatrace.com/api
 
-  # Link to api reference for further information: https://docs.dynatrace.com/docs/ingest-from/setup-on-k8s/reference/dynakube-parameters
+  # Optional: Name of the secret holding the tokens used for connecting to Dynatrace.
+  #
+  # tokens: ""
+
+  # Optional: Defines a custom pull secret in case you use a private registry when pulling images from the Dynatrace environment
+  #
+  # customPullSecret: "custom-pull-secret"
+
+  # Optional: Disable certificate check for the connection between Dynatrace Operator and the Dynatrace Cluster.
+  #
+  # skipCertCheck: false
+
+  # Optional: Set custom proxy settings either directly or from a secret with the field 'proxy'
+  #
+  # proxy:
+  #   value: my-proxy-url.com
+  #   valueFrom: name-of-my-proxy-secret
+
+  # Optional: Add custom RootCAs from a configmap.
+  #
+  # trustedCAs: name-of-my-ca-configmap
+
+  # Optional: Set a network zone for the OneAgent and ActiveGate pods.
+  #
+  # networkZone: name-of-my-network-zone
+
+  # Optional: Configure istio to allow access to the Dynatrace Cluster from the OneAgent or ActiveGate pods.
+  #
+  # enableIstio: true
+
+  # Optional: Minimum minutes between Dynatrace API requests.
+  #
+  # dynatraceApiRequestThreshold: 15
+
+  # Configuration for Metadata Enrichment.
+  #
+  # metadataEnrichment:
+  #  enabled: true
+
+    # Optional: The namespaces where you want Dynatrace Operator to inject
+    #
+    # namespaceSelector:
+    #   matchLabels:
+    #     app: my-app
+    #   matchExpressions:
+    #   - key: app
+    #     operator: In
+  #     values: [my-frontend, my-backend, my-database]
 
   kspm: {}
 
+  # Configuration for ActiveGate instances.
+  #
   activeGate:
-    tlsSecretName: <ag-tls-name>
-
+    # Defines the ActiveGate capabilities
+    #
     capabilities:
       - kubernetes-monitoring
 
-    # The KSPM ActiveGate pipeline is enabled by default with ActiveGate >= 1.311. This custom properties are obsolete then.
+    # Optional: Amount of replicas of ActiveGate pods.
+    #
+    # replicas: 1
+
+    # Optional: Use a custom ActiveGate image
+    #
+    # image: ""
+
+    # Optional: Set the ActiveGate group
+    #
+    # group: ""
+
+    # Optional: Add a custom properties file by providing it as a value or by referencing it from a secret.
```

Comment:
Is this in the case of kspm really optional? If i remember correctly customProperties are required, or am I wrong?

Reply:
yes, [optional starting with AG 311](https://dt-rnd.atlassian.net/browse/DAQ-5712).